plugins/modules/packaging/os/macports.py

```diff
@@ -148,17 +148,18 @@ def query_port(module, port_path, name, state="present"):
 
     if state == "present":
 
-        rc, out, err = module.run_command("%s installed | grep -q ^.*%s" % (shlex_quote(port_path), shlex_quote(name)), use_unsafe_shell=True)
-        if rc == 0:
+        rc, out, err = module.run_command([port_path, "-q", "installed", name])
```

Comment:
:+1: for getting rid of providing the command as a string, using grep and using `use_unsafe_shell=True`!